tests/requestparsing/basicauth_test.go

```diff
@@ -0,0 +1,170 @@
+package requestparsing
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/google/go-safeweb/testing/requesttesting"
+)
+
+func TestBasicAuth(t *testing.T) {
+	type testWant struct {
+		headers  map[string][]string
+		ok       bool
+		username string
+		password string
+	}
```

Comment:
I don't generally have an issue with creating a new struct for `want` values, even though I've mostly seen just a `wantFoo` prefix.

In this case, I think it is a bit problematic, because "ok" doesn't mean much. I'd recommend having this instead:

```
type basicAuth {
  username string
  password string
  ok bool
}
var tests = []struct{
  name string
  request []byte
  wantBasicAuth basicAuth
  wantHeaders map[string][]string
}{...}
```